app/lib/methods/subscriptions/room.ts

```diff
@@ -201,6 +202,62 @@ export default class RoomSubscription {
 					}
 				}
 			});
+		} else if (ev === 'deleteMessageBulk') {
+			InteractionManager.runAfterInteractions(async () => {
+				try {
+					const { rid, excludePinned, ignoreDiscussion, ts, users, ids } = ddpMessage.fields.args[0];
+					const { $gt, $lt, $gte, $lte } = ts || {};
+					const db = database.active;
+
+					const query: Q.Clause[] = [
+						Q.where('rid', rid)
+					];
+
+					if($gt?.$date && $lt?.$date){
+						query.push(
+							Q.where('ts', Q.gt(ts.$gt.$date)),
+							Q.where('ts', Q.lt(ts.$lt.$date))
+						);
+					}
+
+					//only present when inclusive is true in api
+					if($gte?.$date && $lte?.$date){
+						query.push(
+							Q.where('ts', Q.gte(ts.$gte.$date)),
+							Q.where('ts', Q.lte(ts.$lte.$date))
+						);
+					}
+
+					users.forEach((user: string) => {
+						query.push(Q.where('u', Q.like(`%\"username\":\"${user}\",%`)));
+					});
+
+					if (excludePinned) {
+						query.push(
+							Q.or(
+								Q.where('pinned', false),
+								Q.where('pinned', null)
+							)
+						);
+					}
+
+					if (ignoreDiscussion) {
+						query.push(Q.where('drid', null));
+					}
+
+					if(ids){
+						query.push(Q.where('id', Q.oneOf(ids)));
+					}
+
+					const messages = await db.get('messages').query(...query).fetch();
+					
+					await db.write(async () => {
+						await db.batch(...messages.map(message => message.prepareDestroyPermanently()));
+					});
+				} catch (e) {
+					console.log(e);
```

Comment:
Let's log this properly to track issues.
```suggestion
					log(e);
```